internal/services/nginx/nginx_deployment_data_source.go

```diff
@@ -70,6 +70,7 @@ func (m DeploymentDataSource) Attributes() map[string]*pluginsdk.Schema {
 		},
 
 		"managed_resource_group": {
+			// TODO: rename this to `managed_resource_group_name` in 4.0
```

Comment:
Since it's just one property in the resource and data source, instead of leaving this comment here could we just do the rename using the `FourPointOhBeta` flag.